File: src/benchmarks/go/manyrefs.go
package main

func test() func() *[]int {
	var (
		local1 *int
		local2 *int
		local3 *int
		local4 *int
		local5 *int
		local6 *int
		local7 *int
		local8 *int
		local9 *int
	)

	_ = local1
	_ = local2
	_ = local3
	_ = local4
	_ = local5
	_ = local6
	_ = local7
	_ = local8
	_ = local9

	var shared *[]int

	var ret = func() *[]int {
		return shared
	}

	return ret
}

func main() {
	for i := 0; i < 1000000; i++ {
		test()()
	}
}

File: src/benchmarks/c/manyrefs.c
#include <stdlib.h>

typedef int (*f)();

void *shared;

int ret()
{
    return (int)shared;
}

f test()
{
    void *local1 = malloc(4);
    void *local2 = malloc(4);
    void *local3 = malloc(4);
    void *local4 = malloc(4);
    void *local5 = malloc(4);
    void *local6 = malloc(4);
    void *local7 = malloc(4);
    void *local8 = malloc(4);
    void *local9 = malloc(4);

    shared = malloc(1);

    free(local1);
    free(local2);
    free(local3);
    free(local4);
    free(local5);
    free(local6);
    free(local7);
    free(local8);
    free(local9);
    return ret;
}

int main()
{
    for (int i = 0; i < 1000000; i++)
        test()();
    free(shared);
}